stl/inc/generator

```diff
@@ -365,13 +363,35 @@ private:
         };
     };
 
+    _NODISCARD _Nest_info* _Try_get_nest_info() const noexcept {
+        if (_Data & 1U) {
+            return reinterpret_cast<_Nest_info*>(_Data ^ 1U);
+        }
+
+        return nullptr;
+    }
+
+    _NODISCARD coroutine_handle<_Base> _Get_top() const noexcept {
+        _STL_INTERNAL_CHECK((_Data & 1U) == 0);
+        return coroutine_handle<_Base>::from_address(reinterpret_cast<void*>(_Data));
+    }
+
+    void _Set_nest_info(_Nest_info* _Info) noexcept {
+        _Data = reinterpret_cast<uintptr_t>(_Info) | 1U;
+    }
+
+    void _Set_top(coroutine_handle<_Base> _Top) noexcept {
+        _Data = reinterpret_cast<uintptr_t>(_Top.address());
+    }
+
     template <class, class>
     friend struct _Gen_iter_provider;
 
-    // _Top and _Info are mutually exclusive, and could potentially be merged.
-    coroutine_handle<_Base> _Top = coroutine_handle<_Base>::from_promise(*this);
+    // First bit of `_Data` indicates stored information:
+    // 0 - `_Data` is a top coroutine handle,
+    // 1 - `_Data ^ 1U` is a pointer to an object of type `_Nest_info`.

```

Comment:
Really nitpicky, but these dashes kind of look like minus operators during a quick read.